ydb/library/yql/providers/s3/provider/yql_s3_provider.h

```diff
@@ -27,6 +37,8 @@ struct TS3State : public TThrRefBase
     TS3Configuration::TPtr Configuration = MakeIntrusive<TS3Configuration>();
     const NKikimr::NMiniKQL::IFunctionRegistry* FunctionRegistry = nullptr;
     ISecuredServiceAccountCredentialsFactory::TPtr CredentialsFactory;
+
+    TS3FileQueueParams FileQueueParams;
```

Comment:
Does it work with multiple S3 Sources per query?

Reply:
It does. Tested on query with 2 data sources from different buckets.